vector/src/main/java/im/vector/app/features/location/LocationSharingFragment.kt

```diff
@@ -118,8 +130,28 @@ class LocationSharingFragment @Inject constructor(
                 .show()
     }
 
-    override fun invalidate() = withState(viewModel) { state ->
-        views.mapView.render(state.toMapState())
-        views.shareLocationGpsLoading.isGone = state.lastKnownLocation != null
+    private fun initOptionsPicker() {
+        // TODO
+        //  change the options dynamically depending on the current chosen location
+        views.shareLocationOptionsPicker.setOptions(LocationSharingOption.USER_CURRENT)
+        views.shareLocationOptionsPicker.optionPinned.debouncedClicks {
+            viewModel.handle(LocationSharingAction.OnShareLocation)
+        }
+        views.shareLocationOptionsPicker.optionUserCurrent.debouncedClicks {
+            // TODO
+        }
+        views.shareLocationOptionsPicker.optionUserLive.debouncedClicks {
+            // TODO
```

Comment:
What do we plan to do here?

Reply:
For the live location sharing, the app will send location of the user periodically during a certain amount of time until the user stops sharing its location. And in timeline we will also be able to see locations of different several users currently live sharing their location.